chain-impl-mockchain/src/rewards.rs

```diff
@@ -267,6 +349,32 @@ mod tests {
         }
     }
 
+    impl Arbitrary for RewardLimitByStake {
+        fn arbitrary<G: Gen>(gen: &mut G) -> Self {
+            // ugly, but avoids generating unwanted 0.  KH
+            let denominator = u32::arbitrary(gen) + 1;
+            let numerator = u32::arbitrary(gen) % denominator;
+
+            RewardLimitByStake {
+                numerator,
+                denominator: NonZeroU32::new(denominator).unwrap(),
+            }
+        }
+    }
+
+    impl Arbitrary for PoolLimit {
+        fn arbitrary<G: Gen>(gen: &mut G) -> Self {
+            // ugly, but avoids generating unwanted 0.  KH
+            let npools = u32::arbitrary(gen) + 1;
+            let npools_threshold = u32::arbitrary(gen) + 1;
```

Comment:
should be `(u32::arbitrary(gen) % REASONABLE_VALUE) + 1` to avoid the overflow

Reply:
OK.  Will change - was using the same code as in other places.